product_code/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala

```diff
@@ -140,7 +148,14 @@ class CourseServiceImpl(
           throw UC4Exception.PathParameterMismatch
         }
 
-        val validationErrors = updatedCourse.validate
+        val validationErrors = try {
+          Await.result(updatedCourse.validate, 5.seconds)
+        }
+        catch {
+          case _: TimeoutException => throw UC4Exception.ValidationTimeout
+          case e: Exception        => throw e
```

Comment:
This needs to be changed into a Internal Server Error
```suggestion
            case e: Exception        => throw UC4Exception.InternalServerError("Validation Error", e.getMessage)
```